### ISSUE TITLE
metalk8s-solutions ConfigMap is not updated at import

### ISSUE THREAD

**Issue by alexandre-allard** [scality/metalk8s#2344]
**Component**: scripts, solutions

**What happened**:
When importing a Solution, the ConfigMap metalk8s-solutions is not automatically updated (it's only done when (de)activating the Solution) to reflect the new state.

**What was expected**:
The ConfigMap is updated with the newly imported archive.

**Steps to reproduce**:
Bootstrap a node, then:
```
# ./solutions.sh import --archive </path/to/archive.iso>
# kubectl get cm -n metalk8s-solutions metalk8s-solutions -o yaml
apiVersion: v1
kind: ConfigMap
metadata:
  creationTimestamp: "2020-03-31T14:02:54Z"
  labels:
    metalk8s.scality.com/version: 2.4.3-dev
  name: metalk8s-solutions
  namespace: metalk8s-solutions
  resourceVersion: "75106"
  selfLink: /api/v1/namespaces/metalk8s-solutions/configmaps/metalk8s-solutions
  uid: 659c627f-d252-483c-bc72-6ce0e4521095
```
**Resolution proposal** (optional):
Update the ConfigMap after each import and unimport in the `./solutions.sh import` command:
```
diff --git a/scripts/solutions.sh b/scripts/solutions.sh
index 9b6df8da..e3efb207 100755
--- a/scripts/solutions.sh
+++ b/scripts/solutions.sh
@@ -283,6 +283,10 @@ import_solution() {
     run "Importing Solutions" \
         salt_minion_exec state.sls metalk8s.solutions.available \
         saltenv="$SALTENV"
+    run "Updating Solutions ConfigMap" \
+        salt_master_exec salt-run state.orchestrate \
+        metalk8s.orchestrate.solutions.deploy-components \
+        pillar="{'bootstrap_id': '$(get_salt_minion_id)'}"
     run "Configuring Metalk8s registry" \
         salt_minion_exec state.sls metalk8s.repo.installed \
         saltenv="$SALTENV"
@@ -295,6 +299,10 @@ unimport_solution() {
     run "Unimporting Solutions" \
         salt_minion_exec state.sls metalk8s.solutions.available \
         saltenv="$SALTENV"
+    run "Updating Solutions ConfigMap" \
+        salt_master_exec salt-run state.orchestrate \
+        metalk8s.orchestrate.solutions.deploy-components \
+        pillar="{'bootstrap_id': '$(get_salt_minion_id)'}"
     run "Configuring Metalk8s registry" \
         salt_minion_exec state.sls metalk8s.repo.installed \
         saltenv="$SALTENV"
```

**Comment by gdemonet** [comment]
Let's implement it as you suggest for now, but I think it'd be better to wrap the import + CM update in a single orchestrate. This would allow us to perform more complex actions in the future, such as validation of the archive contents, or propagation to other Bootstraps if we had any.

**Comment by alexandre-allard** [comment]
It's just a quickfix, but we'll need to improve this as you suggested, probably during Solution Management iteration 2.

**Comment by alexandre-allard** [comment]
fixed by #2385
